compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2192,123 +2211,46 @@ OMR::ARM64::TreeEvaluator::vreductionFirstNonZeroEvaluator(TR::Node *node, TR::C
-         movOp = TR::InstOpCode::smovwb;
-         break;
-      case TR::Int16:
-         op = isMax ? TR::InstOpCode::vsmaxv8h : TR::InstOpCode::vsminv8h;
-         movOp = TR::InstOpCode::smovwh;
-         break;
-      case TR::Int32:
-         op = isMax ? TR::InstOpCode::vsmaxv4s : TR::InstOpCode::vsminv4s;
-         movOp = TR::InstOpCode::umovws;
-         break;
-      default:
-         TR_ASSERT_FATAL_WITH_NODE(node, false, "Unexpected element type");
-         break;
-      }
-
-   TR::Register *tmpReg = cg->allocateRegister(TR_VRF);
-   TR::Register *resReg = cg->allocateRegister(TR_GPR);
-
-   generateTrg1Src1Instruction(cg, op, node, tmpReg, sourceReg);
-   generateMovVectorElementToGPRInstruction(cg, movOp, node, resReg, tmpReg, 0);
-
-   cg->stopUsingRegister(tmpReg);
-   node->setRegister(resReg);
-   cg->decReferenceCount(sourceChild);
-
-   return resReg;
-   }
-
 /**
  * @brief A helper function for generating instuction sequence for reduction min/max operations for vectors of 64-bit integer elements.
  *
  * @param[in] node: node
  * @param[in] isMax: true if operation is max
+ * @param[in] resReg: result register
+ * @param[in] srcReg: source register
  * @param[in] cg: CodeGenerator
  * @return register containing the result
  */
 static TR::Register *
-vreductionMinMaxInt64Helper(TR::Node *node, bool isMax, TR::CodeGenerator *cg)
+vreductionMinMaxInt64Helper(TR::Node *node, bool isMax, TR::Register *resReg, TR::Register *srcReg, TR::CodeGenerator *cg)
    {
-   TR::Node *sourceChild = node->getFirstChild();
-   TR::Register *sourceReg = cg->evaluate(sourceChild);
-
-   TR_ASSERT_FATAL_WITH_NODE(node, sourceReg->getKind() == TR_VRF, "unexpected Register kind");
-
    TR::Register *tmpReg = cg->allocateRegister(TR_GPR);
-   TR::Register *resReg = cg->allocateRegister(TR_GPR);
 
-   generateMovVectorElementToGPRInstruction(cg, TR::InstOpCode::umovxd, node, tmpReg, sourceReg, 0);
-   generateMovVectorElementToGPRInstruction(cg, TR::InstOpCode::umovxd, node, resReg, sourceReg, 1);
+   generateMovVectorElementToGPRInstruction(cg, TR::InstOpCode::umovxd, node, tmpReg, srcReg, 0);
+   generateMovVectorElementToGPRInstruction(cg, TR::InstOpCode::umovxd, node, resReg, srcReg, 1);
 
    generateCompareInstruction(cg, node, tmpReg, resReg, true);
    generateCondTrg1Src2Instruction(cg, TR::InstOpCode::cselx, node, resReg, tmpReg, resReg, isMax ? TR::CC_GT : TR::CC_LT);
 
    cg->stopUsingRegister(tmpReg);
-   node->setRegister(resReg);
-   cg->decReferenceCount(sourceChild);
 
    return resReg;
    }
 
 /**
- * @brief A helper function for generating instuction sequence for reduction min/max operations for vectors of float elements.
+ * @brief A helper function for generating instuction sequence for reduction max operations for vectors of 64-bit integer elements.
  *
  * @param[in] node: node
- * @param[in] et: element type
- * @param[in] isMax: true if operation is max
+ * @param[in] et: data type
```

Comment:
"element type" and "data type" are mixed for the parameter `et` for helper functions in this file.

Reply:
Thanks for pointing it out. I replaced all occurrences of "data type" with "element type".